### PR TITLE
Spidy test upgrade

### PR DESCRIPTION
**Pull request by influx6**

### CODE REVIEW

File: spidy/spidy.go [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+	pl, err := pool.New("spidy", "collectFrom", poolCfg)
+	if err != nil {
+		fmt.Printf("Spidy failed to create work pool: %s\n", err.Error())
+		close(dead)
+		return
+	}
+
+	status, crawleable, err := evaluatePath(path.String())
+	if err != nil {
+		pl.Shutdown("spidy")
```
Comment (egonzo) [inline spidy/spidy.go]: There is no need for these statements if you move the defer statements on line 89 & 90 above the evaluatePath function call

File: spidy/spidy.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+		dead <- LinkReport{Link: path.String(), Status: status, Error: err}
+		close(dead)
+		return
+	}
+
+	defer pl.Shutdown("spidy")
```
Comment (egonzo) [inline spidy/spidy.go]: Move above the evalutatePath function call

File: spidy/spidy.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+	if err := farmLinks(p.path, links); err != nil {
+		fmt.Printf("Spidy Failed to Farm Links for Page[%s]: Error[%s]\n", p.path, err.Error())
+		p.dead <- LinkReport{Link: p.path, Status: http.StatusInternalServerError, Error: err}
+		return
+	}
+
+	for {
+		select {
+		case <-time.After(time.Second):
```
Comment (egonzo) [inline spidy/spidy.go]: What is the purpose of this timer if all it does is continue? It is not necessary. The case statement below will get triggered when something is received on the channel.

File: vendor/github.com/ardanlabs/kit/cfg/cfg.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+	c := &Config{m: m}
+
+	return c, nil
+}
+
+// Log returns a string to help with logging your configuration. It excludes
+// any values whose key contains the string "PASS".
+func (c *Config) Log() string {
+	c.mu.RLock()
```
Comment (egonzo) [inline vendor/github.com/ardanlabs/kit/cfg/cfg.go]: What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.

File: vendor/github.com/ardanlabs/kit/cfg/cfg.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+		}
+	}
+
+	return buf.String()
+}
+
+// String returns the value of the given key as a string. It will return an
+// error if key was not found.
+func (c *Config) String(key string) (string, error) {
+	c.mu.RLock()
```
Comment (egonzo) [inline vendor/github.com/ardanlabs/kit/cfg/cfg.go]: What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.

File: vendor/github.com/ardanlabs/kit/cfg/cfg.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+		return "", fmt.Errorf("Unknown key %s !", key)
+	}
+
+	return value, nil
+}
+
+// MustString returns the value of the given key as a string. It will panic if
+// the key was not found.
+func (c *Config) MustString(key string) string {
+	c.mu.RLock()
```
Comment (egonzo) [inline vendor/github.com/ardanlabs/kit/cfg/cfg.go]: What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.

File: vendor/github.com/ardanlabs/kit/cfg/cfg.go [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+	{
+		c.m[key] = value
+	}
+	c.mu.RUnlock()
+}
+
+// Int returns the value of the given key as an int. It will return an error if
+// the key was not found or the value can't be converted to an int.
+func (c *Config) Int(key string) (int, error) {
+	c.mu.RLock()
```
Comment (egonzo) [inline vendor/github.com/ardanlabs/kit/cfg/cfg.go]: What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.

File: vendor/github.com/ardanlabs/kit/cfg/cfg.go [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+		return 0, err
+	}
+
+	return iv, nil
+}
+
+// MustInt returns the value of the given key as an int. It will panic if the
+// key was not found or the value can't be converted to an int.
+func (c *Config) MustInt(key string) int {
+	c.mu.RLock()
```
Comment (egonzo) [inline vendor/github.com/ardanlabs/kit/cfg/cfg.go]: What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.

**Comment by egonzo** [comment]
Can we pass the url as command line parameters? Setting them as ENV variables will not work for the intended purpose of spidy. They need to be command line params.

**Comment by influx6** [comment]
Yes, we can set them as commandline parameters also. That was the last commit added.

**Pull request merged**